### PR TITLE
Advertise NAV modes through MSP even when mag is powered off (MR)

### PR DESCRIPTION
**Pull request by shellixyz**
Also checks compass health when arming if any navigation mode is enabled.

### CODE REVIEW

**Comment by digitalentity** [comment]
Since we are looking at this, I suggest we change the way firmware advertises modes. Let's add a MSPv2 API that will also include a flag along with each box mode advertized to indicate if mode is currently supported by hardware or not. What do you think?

**Comment by shellixyz** [comment]
What do you have in mind that would require the MSP clients to know about all the modes even if they are not supported ? To show in the configurator for example the reason why a mode is not available ? I ask because the configurator is already not showing the modes not being supported by the hardware. I probably miss something.

**Comment by digitalentity** [comment]
I'm thinking about situations when hardware is configured but failing. I.e. if compass is set to HMC5883 but was not detected at power up (hardware status - failing).

**Comment by digitalentity** [comment]
Same should be done for GPS and baro

File: src/main/navigation/navigation.c [hunk 1]
```diff
@@ -2485,7 +2487,9 @@ bool navigationBlockArming(void)
         return false;
 
     // Apply extra arming safety only if pilot has any of GPS modes configured
-    if ((isUsingNavigationModes() || failsafeMayRequireNavigationMode()) && !((posControl.flags.estPosStatue >= EST_USABLE) && STATE(GPS_FIX_HOME))) {
+    const bool compassIsHealthyOrNotRequiredNorConfigured = (STATE(FIXED_WING) && !compassIsConfigured()) || (getHwCompassStatus() == HW_SENSOR_OK);
```
Comment (digitalentity) [inline src/main/navigation/navigation.c]: Failing compass is an arming blocker by itself. I think this is not necessary

Comment (shellixyz) [reply src/main/navigation/navigation.c]: I simplified the changes. The newly created `compassIsConfigured()` function is not really needed I'm using `getHwCompassStatus() != HW_SENSOR_NONE` instead.

Also you are right the checks in `updateArmingStatus()` should be enough. If the compass is configured and not calibrated nor healthy arming is prevented. I wasn't sure about that I should have checked. I removed the changes in `navigation.c`.

In the end the changes are very small.

**Comment by shellixyz** [comment]
> I'm thinking about situations when hardware is configured but failing. I.e. if compass is set to HMC5883 but was not detected at power up (hardware status - failing).

So if I understand correctly you would like the configurator to show all the modes but indicate for each unavailable mode the HW reason ?

**Pull request merged**